### PR TITLE
Improve tabs UI for project detail page

### PR DESCRIPTION
**Pull request by AllForNothing**
Env.: https://10.161.79.96:8085

### CODE REVIEW

File: src/portal/src/app/project/project-detail/project-detail.component.ts [hunk 1]
```diff
@@ -131,6 +136,32 @@ export class ProjectDetailComponent implements OnInit {
...
+          }
+      );
+    }
+  }
+
+  ngAfterViewInit() {
+    this.previousWindowWidth = window.innerWidth;
+    setTimeout(() => {
+      this.resetTabsForDownSize();
+    }, 10);
```
Comment (zhoumeina) [inline src/portal/src/app/project/project-detail/project-detail.component.ts]: here 10 is a random number, I think set it to 0 will also work.

File: src/portal/src/app/project/project-detail/project-detail.component.ts [hunk 2]
```diff
@@ -188,10 +219,40 @@ export class ProjectDetailComponent implements OnInit {
...
+      }
+    });
+  }
+  resetTabsForUpSize() {
+    for ( let i = 0; i < this.tabLinkNavList.length; i++) {
+      this.tabLinkNavList[i].tabLinkInOverflow = false;
+    }
+    this.cdf.detectChanges();
+    this.resetTabsForDownSize();
+    this.cdf.detectChanges();
```
Comment (zhoumeina) [inline src/portal/src/app/project/project-detail/project-detail.component.ts]: Looks strange, please try if 1 detectChange will work.

File: src/portal/src/app/project/project-detail/project-detail.component.ts [hunk 3]
```diff
@@ -188,10 +219,40 @@ export class ProjectDetailComponent implements OnInit {
...
+      if (right < SHOW_ELLIPSIS_WIDTH) {
+        this.tabLinkNavList[index].tabLinkInOverflow = true;
+      }
+    });
+  }
+  resetTabsForUpSize() {
+    for ( let i = 0; i < this.tabLinkNavList.length; i++) {
+      this.tabLinkNavList[i].tabLinkInOverflow = false;
+    }
+    this.cdf.detectChanges();
```
Comment (zhoumeina) [inline src/portal/src/app/project/project-detail/project-detail.component.ts]: Need add comment here to indicate why need this detectChange.

Comment (AllForNothing) [reply src/portal/src/app/project/project-detail/project-detail.component.ts]: Fixed

Comment (AllForNothing) [reply src/portal/src/app/project/project-detail/project-detail.component.ts]: > Need add comment here to indicate why need this detectChange.

Fixed

File: src/portal/src/app/project/project-detail/project-detail.component.ts (hunk 2 above)
Comment (AllForNothing) [reply src/portal/src/app/project/project-detail/project-detail.component.ts]: > Looks strange, please try if 1 detectChange will work.

2nd removed

File: src/portal/src/app/project/project-detail/project-detail.component.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/project/project-detail/project-detail.component.ts]: > here 10 is a random number, I think set it to 0 will also work.
Agreed!!!

**Pull request merged**